src/API/Trace/Span.php

```diff
@@ -48,12 +48,21 @@ final public static function wrap(SpanContextInterface $spanContext): SpanInterf
     /** @inheritDoc */
     final public function activate(): ScopeInterface
     {
-        return Context::getCurrent()->withContextValue($this)->activate();
+        $context = Context::getCurrent();
+        if (LocalRootSpan::isLocalRoot($context)) {
+            $context = LocalRootSpan::store($context, $this);
+        }
+
+        return $context->withContextValue($this)->activate();
```

Comment:
Already handled in `::storeContext()`; `$context->withContextValue($this)` calls `$this->storeInContext($context)`.